File: web/src/api/market.ts
import type {
  ExchangeSymbolsPayload,
  KlinePayload,
  KlinesQuery,
  SymbolsQuery,
} from "@/types/public"
import request from "@/utils/request"

const Api = {
  Klines: "/api/market/klines",
  Symbols: "/api/market/symbols",
} as const

export function getKlinesApi(params: KlinesQuery) {
  return request.get<KlinePayload[]>(Api.Klines, { params })
}

export function getExchangeSymbolsApi(params?: SymbolsQuery) {
  return request.get<ExchangeSymbolsPayload>(Api.Symbols, { params })
}

File: core/src/http/handlers/market.rs
use axum::{Json, extract::Query};

use crate::{
    contracts::public::{ExchangeSymbolsPayload, KlinePayload, KlinesQuery, SymbolsQuery},
    error::Result,
    http::response::ApiResponse,
    services::market,
};

pub async fn handle_symbols(
    Query(q): Query<SymbolsQuery>,
) -> Result<Json<ApiResponse<ExchangeSymbolsPayload>>> {
    let payload = market::symbols(q).await?;
    Ok(Json(ApiResponse::success(Some(payload), None)))
}

pub async fn handle_klines(
    Query(q): Query<KlinesQuery>,
) -> Result<Json<ApiResponse<Vec<KlinePayload>>>> {
    let payload = market::klines(q).await?;
    Ok(Json(ApiResponse::success(Some(payload), None)))
}
